### PR TITLE
Changes required for compilation on OSX 

### PR DESCRIPTION
**Pull request by undecided**
Hi,
I was trying to install this, but when I tried to compile test/main.nim, I received various errors.
Firstly, I should confess... these changes were best-guess only. I have absolutely no experience in any of this (part of my motivation for trying nimx) - particularly, I've no idea whether GL_FALSE is actually equivalent to whatever GLenum(0) was attempting to achieve - and I'm guessing that, if it does achieve the desired result now, it potentially won't in the future.

However, I'm submitting since these changes did actually fix things for me and allow me to at least compile the test on OSX 10.10.5 / Nim 0.15.2 / sdl2 with nim wrapper 1.1.

I will try to include the errors that the changes fix below, in case you know of any better fixes.

Best regards,
Matthew

P.S: I've noticed that the CI tests are failing - I'll have a look, there may be something odd with my env.

### CODE REVIEW

File: nimx/image.nim [hunk 1]
```diff
@@ -83,7 +83,7 @@ when not defined js:
             of 2: GL_LUMINANCE_ALPHA
             of 3: GL_RGB
             of 4: GL_RGBA
-            else: GLenum(0)
+            else: GL_FALSE
```
Comment (undecided) [inline nimx/image.nim]: This fixes the following error:
`/Users/matthew/.nimble/pkgs/nimx-0.1/nimx/image.nim(86, 25) Error: type mismatch: got (GLenum) but expected 'int literal(6406)'`

File: nimx/cursor.nim [hunk 2]
```diff
@@ -7,6 +7,7 @@ when defined(js) or defined(emscripten):
 elif defined(appKit):
     import private.objc_appkit
     enableObjC()
+    import sdl2
```
Comment (undecided) [inline nimx/cursor.nim]: This fixes the following error:
`/Users/matthew/.nimble/pkgs/nimx-0.1/nimx/cursor.nim(70, 42) Error: undeclared identifier: 'SystemCursor'`

**Comment by yglukhov** [comment]
@undecided, it looks a lot like you've got old version of opengl. Please try to update it. `nimble install -y opengl` and tell me how it goes. Nimx should compile perfectly fine on windows, linux and macosx provided latest versions of its dependencies.

**Comment by undecided** [comment]
Aha! Thank you, yes, that's what I was missing. Well, don't I feel like a wally. Thank you for your help!

**Comment by yglukhov** [comment]
No problem at all. Does it compile without errors now?

**Comment by undecided** [comment]
Oddly, I'm struggling to get it to fail now - I think nimble is doing something odd. But pretty sure that setting the min opengl versions will fix it - so thank you for your lightning responses!

**Comment by yglukhov** [comment]
Thanks for your report as well. :)

**Comment by undecided** [comment]
My pleasure - learned a lot!

**Pull request closed without merging**